widget/sliderswitch.go

```diff
@@ -0,0 +1,58 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/widget"
+)
+
+var _ fyne.Tappable = (*SliderSwitch)(nil)
+
+// SliderSwitch is an extension of a slider that switches
+// between two values when clicked.
+type SliderSwitch struct {
+	widget.Slider
+
+	// OnToggle is called AFTER the switch has changed state.
+	// Bool is true if the switch is in the "ON" state.
+	OnToggle func(bool)
+	// If set, the slider will be moved to an intermediate state,
+	// the function called, and then moved to its final state after
+	// the function returns. Bool is true if the switch is transitioning
+	// to the "ON" state.
+	OnTransition func(bool)
```

Comment:
I don't understand what transitioning is and/or why value 1 between 0 and 2...
For one thing how can you possibly know that 0->1 then follows with 2, it could be 0->1->0.

Reply:
Don't folk normally set the state and then disable to show that it is not available then re-enable when completed?
I wouldn't be exposing transitioning in this way - users can slide to the intermediate position. Also taking this model means the UI will seem to "stick" part way through, looking unresponsive - but it can still be interacted with. It feels untidy I guess.

Reply:
Fair points